test/unit/ngsiv2/provisioning/device-update-registration_test.js

```diff
@@ -0,0 +1,360 @@
+                        type: 'centigrades'
+                    }
+                ],
+                active: [
+                    {
+                        name: 'pressure',
+                        type: 'Hgmm'
+                    }
+                ],
+                service: 'smartGondor',
+                subservice: 'gardens'
+            },
+            'Termometer': {
+                commands: [],
+                lazy: [
+                    {
+                        name: 'temp',
+                        type: 'kelvin'
+                    }
+                ],
+                active: [
+                ],
+                service: 'smartGondor',
+                subservice: 'gardens'
+            }
+        },
+        service: 'smartGondor',
+        subservice: 'gardens',
+        providerUrl: 'http://smartGondor.com',
+        deviceRegistrationDuration: 'P1M',
+        throttling: 'PT5S'
+    },
+    device1 = {
+        id: 'light1',
+        type: 'Light',
+        service: 'smartGondor',
+        subservice: 'gardens',
+    },
+    deviceUpdated = {
+        id: 'light1',
+        type: 'Light',
+        name: 'light1',
+        service: 'smartGondor',
+        subservice: 'gardens',
+        internalId: 'newInternalId',
+        lazy: [
+            {
+                name: 'pressure',
+                type: 'Hgmm'
+            }
+        ],
+        active: [
+            {
+                name: 'temperature',
+                type: 'centigrades'
+            }
+        ]
+    },
+    deviceCommandUpdated = {
+        id: 'light1',
+        type: 'Light',
+        name: 'light1',
+        service: 'smartGondor',
+        subservice: 'gardens',
+        internalId: 'newInternalId',
+        commands: [
+            {
+                name: 'move',
+                type: 'command'
+            }
+        ],
+        active: [
+            {
+                name: 'temperature',
+                type: 'centigrades'
+            }
+        ]
+    },
+    unknownDevice = {
+        id: 'rotationSensor4',
+        type: 'Rotation',
+        name: 'Rotation4',
+        service: 'dumbMordor',
+        subservice: 'gardens',
+        internalId: 'unknownInternalId',
+
+        lazy: [],
+        active: []
+    };
+
+describe('IoT Agent Device Update Registration', function() {
+    beforeEach(function(done) {
+        delete device1.registrationId;
+        logger.setLevel('FATAL');
+
+        nock.cleanAll();
+
+        contextBrokerMock = nock('http://192.168.1.1:1026')
+            .matchHeader('fiware-service', 'smartGondor')
+            .matchHeader('fiware-servicepath', 'gardens')
+                .post('/v2/registrations')
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+        contextBrokerMock
+            .matchHeader('fiware-service', 'smartGondor')
+            .matchHeader('fiware-servicepath', 'gardens')
+            .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5